android/gitudy/presentation/src/main/java/com/takseha/presentation/ui/login/LoginWebViewActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.takseha.presentation.ui.login
+
+import android.content.Intent
+import android.net.Uri
+import android.net.UrlQuerySanitizer
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.takseha.common.model.SharedPreferencesKey
+import com.takseha.common.util.SharedPreferences
+import com.takseha.data.dto.TokenResponse
+import com.takseha.data.repository.Repository
+import com.takseha.presentation.R
+import com.takseha.presentation.databinding.ActivityLoginWebviewBinding
+import com.takseha.presentation.viewmodel.GetTokenViewModel
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+
+class LoginWebViewActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginWebviewBinding
+    private lateinit var viewModel: GetTokenViewModel
+    private lateinit var prefs: SharedPreferences
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login_webview)
+        setBinding()
+
+        viewModel = ViewModelProvider(this)[GetTokenViewModel::class.java]
+        prefs = SharedPreferences(applicationContext)
+
+        binding.loginWebView.run {
+            webViewClient = LoginWebViewClient()
+
+            // webview 사용 로그인을 위해 우회
+            settings.apply {
+                userAgentString = "Mozilla/5.0 AppleWebKit/535.19 Chrome/56.0.0 Mobile Safari/535.19"
+                javaScriptEnabled = true
+            }
+
+            loadUrl(intent.getStringExtra("url").toString())
+        }
+    }
+
+    private fun setBinding() {
+        binding = ActivityLoginWebviewBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+    }
+
+    private fun saveAllToken(url: String?) {
+        val sanitizer = UrlQuerySanitizer()
+        sanitizer.allowUnregisteredParamaters = true;
+        sanitizer.parseUrl(url)
+
+        val platformType = intent.getStringExtra("platformType").toString()
+        val code = sanitizer.getValue("code")
+        val state = sanitizer.getValue("state")
+
+        viewModel.getAllTokens(platformType, code, state)
+
+        viewModel.accessToken.observe(this, Observer {
+            prefs.savePref(
+                SharedPreferencesKey.ACCESS_TOKEN,
+                it
+            )
+        })
+
+        viewModel.refreshToken.observe(this, Observer {
+            prefs.savePref(
+                SharedPreferencesKey.REFRESH_TOKEN,
+                it
+            )
+        })
```

Comment:
```suggestion
        viewModel.accessToken.observe(this, Observer {
            prefs.savePref(
                SharedPreferencesKey.ACCESS_TOKEN,
                it
            )
        })

        viewModel.refreshToken.observe(this, Observer {
            prefs.savePref(
                SharedPreferencesKey.REFRESH_TOKEN,
                it
            )
        })
```

이 토큰을 저장하는 로직은 GetTokenViewModel에 넣는게 더 자연스러워 보여요!

나중에 재발급하는 로직에도 충분히 재사용할 수 있을 것 같아요